src/RemoteResourceController.js

```diff
@@ -26,6 +29,55 @@ module.exports = class RemoteResourceController extends BaseDownloadController {
     super(params);
   }
 
+  async added() {
+    let requests = objectPath.get(this.data, ['object', 'spec', 'requests'], []);
+    let newRequests = [];
+    for (var i = 0; i < requests.length; i++) {
+      let req = requests[i];
+      let reqOpt = clone(req.options);
+      let optional = req.optional || false;
```

Comment:
```suggestion
    const requests = objectPath.get(this.data, ['object', 'spec', 'requests'], []);
    const newRequests = [];
    for (let i = 0; i < requests.length; i++) {
      const req = requests[i];
      const reqOpt = clone(req.options);
      const optional = req.optional || false;
```

Reply:
some get changed so didn't make all const

Reply:
`const` only applies to the variable. The properties of the object the variable points to can still be changed